client/gutenberg/extensions/hello-dolly/hello-block.js

```diff
@@ -0,0 +1,38 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import Badge from 'components/badge';
+import Button from 'components/button';
+import Notice from 'components/notice';
+
+import 'components/badge/style.scss';
+import 'components/button/style.scss';
+import 'components/notice/style.scss';
```

Comment:
Generating CSS from packages is new in Gutenberg as well but we're doing it differently.

Instead of importing "sass" files in JavaScript files, we have an `index.scss` entry point where we import the styles. The styles still live next to the components though.

I like the collocating and the idea that you can import everything directly in the component but for modules I think it's a bit misleading when you read the source code. All what these imports are doing is extract and concat all the stylesheets together (no real scoping which can be implied from imports).

I guess it also relates to the difference between using `webpack` and `sass` + `babel`. We also didn't want to bundle the "package" in a single script and leave it to the package consumer to bundle/tree-shake etc... I guess the use-case here is a bit different because we do want a bundle to be used in JetPack/Calypso/Wp-Admin or maybe we need both in two steps ? An unbundled version  + a bundled version?

Reply:
> I don't like magical systems where the presence of a file changes the behavior of the build without any explicit explanation why.

The `import` seems a bit magical too, tho. (Like Riad said, it's just implied concatenation during the build phase -- but we don't semantically use those 'imported' symbols in our code.)

Reply:
> it's just implied concatenation during the build phase

I think we should discuss this in Slack or in a video call. I'm not sure I am making the connection especially since `import` is part of the language and explicitly calls out the imported files. That is, if you understand Node's semantics, then I'm having trouble understanding why this is magical.